esphome/components/i2c/i2c_bus_esp_idf.cpp

```diff
@@ -273,10 +274,14 @@ void IDFI2CBus::recover_() {
     // When SCL is kept LOW at this point, we might be looking at a device
     // that applies clock stretching. Wait for the release of the SCL line,
     // but not forever. There is no specification for the maximum allowed
-    // time. We'll stick to 500ms here.
-    auto wait = 20;
+    // time. We yield and reset the WDT, so as to avoid triggering reset.
+    // No point in trying to recover the bus by forcing a uC reset. Bus
+    // should recover in a few ms or less else not likely to recovery at
+    // all.
+    auto wait = 250;
     while (wait-- && gpio_get_level(scl_pin) == 0) {
-      delay(25);
+      esp_task_wdt_reset();
```

Comment:
I know this is idf only, but the function actually does a little bit more.
```suggestion
      App.feed_wdt();
```